app/models/petition.rb

```diff
@@ -28,6 +28,7 @@ class Petition < ActiveRecord::Base
 
   has_perishable_token called: 'sponsor_token'
 
+  before_save :stamp_parliament_response_at, if: -> { response.present? && response_was.nil? }
   after_create :set_petition_on_creator_signature
```

Comment:
ActiveRecord has a feature called "dirty tracking" where it knows what attributes were changed during the last save.  See http://api.rubyonrails.org/classes/ActiveModel/Dirty.html for more details at the API level and [this railscast](http://railscasts.com/episodes/109-tracking-attribute-changes) (a bit old but covers the basics of the API)